File: helloworld.js
import fs from 'fs';
import { CID } from 'multiformats/cid'
import { encode, decode, code } from 'multiformats/codecs/raw'
import { sha256 } from 'multiformats/hashes/sha2'
import * as dagPB from '@ipld/dag-pb'

const filePath = './helloworld';
let value = await fs.readFileSync(filePath)
let data = await encode(value)
const hash = await sha256.digest(data)
const cidv0 = CID.create(0, dagPB.code, hash)
const cidv1 = CID.create(1, code, hash)


console.log("\nhash")
console.log(hash)
console.log("\nvalue")
console.log(value)
console.log("\ndata")
console.log(data)
console.log("\ncode")
console.log(code)
console.log("\ncidv0")
console.log(cidv0)
console.log("\ncidv1")
console.log(cidv1)

File: helloworld.py
from multiformats import CID
from multiformats import multibase
from multiformats import multihash

with open('helloworld', 'rb') as file:
    raw_bytes = file.read()
encoded = multibase.encode(raw_bytes, "base32")
digest = multihash.digest(raw_bytes, "sha2-256")
cidv0 = CID("base58btc", 0, "dag-pb", digest)
cidv1 = CID("base32", 1, "raw", digest)

print("\nraw_bytes")
print(raw_bytes)
print("\ndigest")
print(digest)
print("\ncidv0")
print(cidv0)
print("\ncidv1")
print(cidv1)